pallets/pallet-dip-provider/src/benchmarking.rs

```diff
@@ -0,0 +1,108 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use crate::{traits::IdentityProvider, Call, Config, Pallet};
+use frame_benchmarking::v2::*;
+use kilt_support::{
+	benchmark::IdentityContext,
+	traits::{GenerateBenchmarkOrigin, GetWorstCase, Instanciate},
+};
+
+#[benchmarks(
+	where
+		T::CommitOriginCheck: GenerateBenchmarkOrigin<T::RuntimeOrigin, T::AccountId, T::Identifier>,
+		T::AccountId: Instanciate,
+		T::Identifier: Instanciate,
+		<<T as Config>::IdentityProvider as IdentityProvider<T>>::Success: GetWorstCase<IdentityContext<T::Identifier, T::AccountId>>
+)]
+mod benchmarks {
+
+	type IdentityContextOf<Runtime> =
+		IdentityContext<<Runtime as Config>::Identifier, <Runtime as frame_system::Config>::AccountId>;
+
+	use super::*;
+
+	#[benchmark]
+	fn commit_identity() {
+		let submitter = T::AccountId::new(1);
+		let subject = T::Identifier::new(1);
+		let commitment_version = 0;
+
+		let context = IdentityContext::<T::Identifier, T::AccountId> {
+			did: subject.clone(),
+			submitter: submitter.clone(),
+		};
+
+		assert!(Pallet::<T>::identity_commitments(&subject, commitment_version).is_none());
+
+		let origin: T::RuntimeOrigin = T::CommitOriginCheck::generate_origin(submitter, subject.clone());
+
+		<<<T as Config>::IdentityProvider as IdentityProvider<T>>::Success as GetWorstCase<
```

Comment:
Especially if the logic to take deposits is moved **outside** of the trait impl (as it should be), I think you need to capture this value and use its components to do all the setup steps you need.

Reply:
Perhaps I misunderstood something. Is the provider pallet designed exclusively for KILT? If not, I cannot proceed, as other identity chains will likely have different data structures, and I have no idea of their formats here.

The worst case is generated so that the user can set up the worst-case scenario for a particular identity context. This is a general approach so that any identity chain can use these benchmarks.